From 5fd19044b6fb63e3f9195c5ffa3c7b78183f24f5 Mon Sep 17 00:00:00 2001
From: Anantha Krishnan <ananthk@codeaurora.org>
Date: Tue, 26 Aug 2014 04:03:53 +0530
Subject: [PATCH 11/25] bluetooth : Configure the Tx idle timeout to 1sec

when there is no activity on the host tx for 1 sec
indicate the controller to enter into sleep.
1sec is the optimal value we need to satisfy the power
and performance requirements.

Change-Id: I9c2a2973a7fbb1a49460e06117c264aa4e0e44e4
Signed-off-by: Anantha Krishnan <ananthk@codeaurora.org>
---
 drivers/bluetooth/hci_ibs.c |   11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

diff --git a/drivers/bluetooth/hci_ibs.c b/drivers/bluetooth/hci_ibs.c
index 42a67ab..55d1b95 100644
--- a/drivers/bluetooth/hci_ibs.c
+++ b/drivers/bluetooth/hci_ibs.c
@@ -63,6 +63,9 @@
 #define HCI_IBS_WAKE_IND	0xFD
 #define HCI_IBS_WAKE_ACK	0xFC
 
+/* TX idle time out value */
+#define TX_IDLE_TO		1000
+
 /* HCI_IBS receiver States */
 #define HCI_IBS_W4_PACKET_TYPE	0
 #define HCI_IBS_W4_EVENT_HDR	1
@@ -270,7 +273,7 @@ static void ibs_wq_awake_device(struct work_struct *work)
 	ibs->ibs_sent_wakes++; /* debug */
 
 	/* start retransmit timer */
-	mod_timer(&ibs->wake_retrans_timer, jiffies + wake_retrans);
+	mod_timer(&ibs->wake_retrans_timer, jiffies + msecs_to_jiffies(10));
 
 	spin_unlock_irqrestore(&ibs->hci_ibs_lock, flags);
 
@@ -656,7 +659,8 @@ static void ibs_device_woke_up(struct hci_uart *hu)
 			skb_queue_tail(&ibs->txq, skb);
 		/* switch timers and change state to HCI_IBS_TX_AWAKE */
 		del_timer(&ibs->wake_retrans_timer);
-		mod_timer(&ibs->tx_idle_timer, jiffies + tx_idle_delay);
+		mod_timer(&ibs->tx_idle_timer, jiffies +
+			msecs_to_jiffies(TX_IDLE_TO));
 		ibs->tx_ibs_state = HCI_IBS_TX_AWAKE;
 	}
 
@@ -686,7 +690,8 @@ static int ibs_enqueue(struct hci_uart *hu, struct sk_buff *skb)
 	case HCI_IBS_TX_AWAKE:
 		BT_DBG("device awake, sending normally");
 		skb_queue_tail(&ibs->txq, skb);
-		mod_timer(&ibs->tx_idle_timer, jiffies + tx_idle_delay);
+		mod_timer(&ibs->tx_idle_timer, jiffies +
+			msecs_to_jiffies(TX_IDLE_TO));
 		break;
 
 	case HCI_IBS_TX_ASLEEP:
-- 
1.7.9.5

